src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -1088,18 +1092,37 @@ ActionResultMetadata parseActionResultMetadata(
           new FileMetadata(localPath, outputFile.getDigest(), outputFile.getIsExecutable()));
     }
 
-    ImmutableMap.Builder<Path, SymlinkMetadata> symlinks = ImmutableMap.builder();
-    Iterable<OutputSymlink> outputSymlinks =
-        Iterables.concat(result.getOutputFileSymlinks(), result.getOutputDirectorySymlinks());
-    for (OutputSymlink symlink : outputSymlinks) {
-      Path localPath =
-          remotePathResolver.outputPathToLocalPath(encodeBytestringUtf8(symlink.getPath()));
-      symlinks.put(
-          localPath, new SymlinkMetadata(localPath, PathFragment.create(symlink.getTarget())));
+    Map<Path, SymlinkMetadata> symlinkMap = new HashMap<>();
+    var symlinkLists =
+        ImmutableList.of(
+            result.getOutputFileSymlinks(),
+            result.getOutputDirectorySymlinks(),
+            result.getOutputSymlinks());
+    for (var list : symlinkLists) {
+      for (OutputSymlink symlink : list) {
```

Comment:
There's no need for a nested loop; just do an `Iterables.concat` of the three lists and iterate over that.

Reply:
Thanks! The result seems a lot cleaner 🫡